File: 448.Find All Numbers Disappeared in an Array/solution.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for(int i=0;i<nums.length;i++){
            int curr = Math.abs(nums[i]);
            if(nums[curr-1]>0){
                nums[curr-1] *=-1;
            }
        }
        List<Integer> dis = new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            if(nums[i]>0){
                dis.add(i+1);
            }
        }
        return dis;
    }
}

File: 448.Find All Numbers Disappeared in an Array/solution.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        res = []

        record = {i:0 for i in range(1,n+1)}

        for i in nums:
            record[i] = 1
        
        for i in range(1,n+1):
            if record[i]==0:
                res.append(i)
        
        return res